src/shogun/metric/LMNN.cpp

```diff
@@ -109,8 +104,13 @@ void CLMNN::train(SGMatrix<float64_t> init_transform)
 
 		// Compute the objective, trace of Mahalanobis distance matrix (L squared) times the gradient
 		// plus the number of current impostors to account for the margin
+		// trace of the product of two matrices computed fast using
+		// trace(A*B)=sum(A.*B')
 		SG_DEBUG("Computing objective.\n")
-		obj[iter] = TRACE(L.transpose()*L,gradient) + m_regularization*cur_impostors.size();
+		obj[iter] = m_regularization * cur_impostors.size();
```

Comment:
but this is slower than the old version :(
Maybe introduce this macro as a function in linalg? `trace_dot` or so?